src/main/java/org/opensearch/plugin/insights/core/service/TopQueriesService.java

```diff
@@ -109,6 +113,13 @@ public class TopQueriesService {
         topQueriesStore = new PriorityQueue<>(topNSize, (a, b) -> SearchQueryRecord.compare(a, b, metricType));
         topQueriesCurrentSnapshot = new AtomicReference<>(new ArrayList<>());
         topQueriesHistorySnapshot = new AtomicReference<>(new ArrayList<>());
+        queryGroupingService = new QueryGroupingService(
+            metricType,
+            QueryInsightsSettings.DEFAULT_GROUPING_TYPE,
+            AggregationType.AVERAGE,
+            topQueriesStore,
+            topNSize
+        );
```

Comment:
I don't think we should be calling this `QueryGroupingService`. Service generally acts independently through well defined interaction points. Maybe `QueryGrouper` is more appropriate term.